js/models.js

```diff
@@ -83,46 +88,60 @@ export function setModelDefaults(model) {
   if (config._button._isFullWidth) {
     config._stepLocking._isEnabled = true;
   }
-  setModelConfig(model, config);
+  return config;
 }
 
 /**
- * Returns the specified or derived trickle configuration for the given model
- * @param {Backbone.Model} model
- * @returns {Object}
+ * Returns an array of models from which to derive a trickle config or null
+ * @param {Backbone.Model} configModel
+ * @returns {[Backbone.Model]|null}
  */
-export function getModelConfig(model) {
-  const inheritedConfigId = model.get('_trickleConfigId');
-  if (inheritedConfigId) {
-    const inheritedConfig = Adapt.findById(inheritedConfigId).get('_trickle');
-    const localConfig = model.get('_trickle') || {};
-    // Introduced for the AAT whereby explicit inheritance is required
-    const isInheritedOnly = model.get(localConfig._isInherited);
-    const localOverrides = isInheritedOnly ? {} : localConfig;
-    const derivedConfig = $.extend(true, {}, inheritedConfig, localOverrides);
-    return derivedConfig;
+export function getModelInheritanceChain(configModel) {
+  if (!data.isReady) throw new Error('Trickle cannot resolve inheritance chains until data is ready');
+  const type = configModel.get('_type');
+  if (type === 'block') {
+    return [ configModel, configModel.getParent() ].filter(ancestor => {
+      const config = ancestor.get('_trickle');
+      // Remove models with no config and models which explicitly require inheritance
+      return (config && !config._isInherited);
+    });
   }
-  return model.get('_trickle');
-}
+  if (type === 'article') {
+    return [ configModel ];
+  }
+  return null;
+};
 
 /**
- * Set the specified trickle configuration for the given model
+ * Returns the derived trickle configuration for the given model
  * @param {Backbone.Model} model
- * @param {object} config
+ * @returns {Object}
  */
-export function setModelConfig(model, config) {
-  model.unset('_trickleConfigId');
-  model.set('_trickle', config);
+export function getModelConfig(model) {
+  const inheritance = getModelInheritanceChain(model);
+  if (!inheritance?.length) return null;
+  if (isModelArticleWithOnChildren(model)) return null;
```

Comment:
could combine these 2 if statements to return null.
`if (!inheritance?.length || isModelArticleWithOnChildren(model)) return null;`